packages/vite/src/node/config.ts

```diff
@@ -120,15 +138,142 @@ export function defineConfig(config: UserConfigExport): UserConfigExport {
   return config
 }
 
-export type PluginOption =
-  | Plugin
-  | false
-  | null
-  | undefined
-  | PluginOption[]
-  | Promise<Plugin | false | null | undefined | PluginOption[]>
+export interface DevEnvironmentOptions {
+  /**
+   * Files tßo be pre-transformed. Supports glob patterns.
+   */
+  warmup?: string[]
+  /**
+   * Pre-transform known direct imports
+   * defaults to true for the client environment, false for the rest
+   */
+  preTransformRequests?: boolean
+  /**
+   * Enables sourcemaps during dev
+   * @default { js: true }
+   * @experimental
+   */
+  sourcemap?: boolean | { js?: boolean; css?: boolean }
+  /**
+   * Whether or not to ignore-list source files in the dev server sourcemap, used to populate
+   * the [`x_google_ignoreList` source map extension](https://developer.chrome.com/blog/devtools-better-angular-debugging/#the-x_google_ignorelist-source-map-extension).
+   *
+   * By default, it excludes all paths containing `node_modules`. You can pass `false` to
+   * disable this behavior, or, for full control, a function that takes the source path and
+   * sourcemap path and returns whether to ignore the source path.
+   */
+  sourcemapIgnoreList?:
+    | false
+    | ((sourcePath: string, sourcemapPath: string) => boolean)
+
+  /**
+   * Optimize deps config
+   */
+  optimizeDeps?: DepOptimizationConfig
```

Comment:
It looks like currently `optimizeDeps.entries` is not exposed (`entries` is in `DepOptimizationOptions` instead of `DepOptimizationConfig`). Is this intended?

```ts
  environments: {
    client: {
      dev: {
        optimizeDeps: {
          // this is currently type error
          entries: []
        },
      },
```

Reply:
Could you expand on a good use case for having different `entries`? These are only used while scanning, that is currently only enabled for the `client` environment. This is now an artificial limitation though. Given that we can properly issue a reload for all environments, we could opt-in into the full scanning + registering missing dependencies instead of a static list if we want.

Reply:
Right, what I meant is only for `client` environment, just something equivalent of current root `optimizeDeps.entries` to help discovering deps early using initial esbuild scan.

Reply:
The current `config.optimizeDeps.entries` will be used during scan. But I think we should move it to environment options 👍🏼